cmake/modules/boards.cmake

```diff
@@ -94,9 +100,12 @@ if(DEFINED ZEPHYR_BOARD_ALIASES)
 endif()
 
 include(${ZEPHYR_BASE}/boards/deprecated.cmake)
-if(${BOARD}_DEPRECATED)
+if(${BOARD}${BOARD_IDENTIFIER}_DEPRECATED)
   set(BOARD_DEPRECATED ${BOARD} CACHE STRING "Deprecated board name, provided by user")
-  parse_board_components(${BOARD}_DEPRECATED BOARD BOARD_DEPRECATED_REVISION BOARD_DEPRECATED_IDENTIFIER)
+  parse_board_components(
+    ${BOARD}${BOARD_IDENTIFIER}_DEPRECATED
+    BOARD BOARD_DEPRECATED_REVISION BOARD_IDENTIFIER
+  )
```

Comment:
afaict, this will not work for the case where we just want to rename (deprecate) a v2 board, while still support a board to board rename, for example `plank` being a `v2` board supporting identifiers, a use-case also discussed in https://github.com/zephyrproject-rtos/zephyr/pull/68940#discussion_r1492393134.
```
set(plank_DEPRECATED plonk)
``` 

would have all identifiers being passed on.
This use case is different from:
```
set(plank/<identifier>_DEPRECATED plonk)
set(plank_DEPRECATED plonk/<identifier>)
set(plank/<identifier>_DEPRECATED plonk/<other-identifier>)
```
(where one of the sides is containing a `/`)

For the `set(plank_DEPRECATED plonk)` all identifiers must be mapped.

An acceptable limitation, but maybe worth mentioning.

Reply:
That's correct. For this part of your proposal:

> Question then becomes, how to handle a pure board rename, where you don't want to create an entry for each identifier, like a `plank` --> `plonk` rename.
> 
> That could be handled as:
> 
> ```
> set(plank_DEPRECATED plonk) # Pure rename on board level, any identifier applied by user will be transfered.
> ```
> 
> So that `plank/foo` --> `plonk/foo`.

I responded with:

> > Question then becomes, how to handle a pure board rename, where you don't want to create an entry for each identifier, like a `plank` --> `plonk` rename.
> 
> If it were consistent with your overall approach, it would have to be like this:
> 
> ```cmake
> set(plank_DEPRECATED             plonk)
> set(plank/soc_DEPRECATED         plonk/soc)
> set(plank/soc/cpu0_DEPRECATED    plonk/soc/cpu0)
> set(plank/soc/cpu0/ns_DEPRECATED plonk/soc/cpu0/ns)
> set(plank/soc/cpu1_DEPRECATED    plonk/soc/cpu1)
> ```
> 
> I think there's no reason to treat the top level identifier differently from any other level.

and you seemed to agree:

> > If it were consistent with your overall approach, it would have to be like this:
> 
> Correct, if what I wanted to show was all the lines.

So, not sure if there's a contradiction or misunderstanding, but the limitation is intentional.